config/config_params.go

```diff
@@ -102,6 +102,14 @@ type Config struct {
 	UseInternalDataplaneDriver bool   `config:"bool;true"`
 	DataplaneDriver            string `config:"file(must-exist,executable);calico-iptables-plugin;non-zero,die-on-fail,skip-default-validation"`
 
+	// Wireguard configuration
+	WireguardEnabled             bool   `config:"bool;false"`
+	WireguardListeningPort       int    `config:"int;51902"`
+	WireguardRoutingRulePriority int    `config:"int;99"`
+	WireguardRoutingTableIndex   int    `config:"int;99"`
```

Comment:
Please sync up on config model with @neiljerram and @song-jiang since their feature also needs routing tables.  Might be best to have a general config parameter that says "calico is allowed to use this range of routing tables" and then to allocate from that internally.  Then we don't need validation that tells the user "this wireguard table index clashes with that other feature table index range"